### PR TITLE
refactor: Use f-strings everywhere

### PR DESCRIPTION
**Pull request by bdrung**
Use [flynt](https://github.com/ikamensh/flynt) to convert the code from old `%`-formatted and `.format(...)` strings into Python 3.6+'s "f-strings". Not all instances were converted by flynt. The remaining ones were converted manually.

### CODE REVIEW

**Comment by bdrung** [comment]
IMO we could ignore codecov/patch and codecov/project for this merge request.

File: apport/crashdb.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def init_duplicate_db(self, path):
         cur.execute("PRAGMA integrity_check")
         result = cur.fetchall()
         if result != [("ok",)]:
-            raise SystemError("Corrupt duplicate db:" + str(result))
+            raise SystemError(f"Corrupt duplicate db:{str(result)}")
```
Comment (schopin-pro) [inline apport/crashdb.py:126]: suggestion (not-blocking): get rid of superfluous `str()` call

The f-string machinery will call it for you :)

```suggestion
            raise SystemError(f"Corrupt duplicate db:{result}")
```

File: apport/crashdb.py [hunk 2]
```diff
@@ -961,9 +959,7 @@ def get_crashdb(auth_file, name=None, conf=None):
                         )
                 except Exception as error:  # pylint: disable=broad-except
                     # ignore broken files
-                    sys.stderr.write(
-                        "Invalid file %s: %s\n" % (cfpath, str(error))
-                    )
+                    sys.stderr.write(f"Invalid file {cfpath}: {str(error)}\n")
```
Comment (schopin-pro) [inline apport/crashdb.py:962]:
```suggestion
                    sys.stderr.write(f"Invalid file {cfpath}: {error}\n")
```

File: apport/crashdb_impl/launchpad.py [hunk 3]
```diff
@@ -109,9 +109,8 @@ def __init__(self, auth, options):
             lp_instance = options.get("launchpad_instance")
             if lp_instance:
                 auth = (
-                    default_credentials_path
-                    + "."
-                    + lp_instance.split("://", 1)[-1]
+                    f"{default_credentials_path}"
+                    f".{lp_instance.split('://', 1)[-1]}"
                 )
```
Comment (schopin-pro) [inline apport/crashdb_impl/launchpad.py:114]: praise: TIL you can use this form of concat on f-strings. Neat!

suggestion (not-blocking): I'd bypass string formatting completely on that one

```python3
                auth = ".".join((
                    default_credentials_path,
                    lp_instance.split("://", 1)[-1],
                ))
```

Less sigils, and puts the separating dot in evidence.
I'd have put this in a proper GH suggestion, but apparently it's not supported on "deleted lines". Oh well.

**Review by schopin-pro — APPROVED** [review]
Nice work! A couple of nitpicks, feel free to disregard them.

File: apport/crashdb_impl/launchpad.py (hunk 3 above)
Comment (bdrung) [reply apport/crashdb_impl/launchpad.py:114]: IMO both variant are equally readable. So I included your suggestion.

**Comment by bdrung** [comment]
I suggest to ignore the failing codecov checks for this change. I am not sure why codecov/project is failing (maybe because the number of lines reduced).

**Pull request merged**